src/app/clusters/time-synchronization-server/time-synchronization-cluster.cpp

```diff
@@ -173,23 +130,29 @@ static bool emitDSTStatusEvent(EndpointId ep, bool dstOffsetActive)
     return true;
 }
 
-static bool emitTimeZoneStatusEvent(EndpointId ep)
+bool emitTimeZoneStatusEvent(TimeSynchronizationCluster * timeSynchronization, EndpointId ep,
+                             DataModel::EventsGenerator * eventsGenerator)
 {
-    const auto & tzList = TimeSynchronizationServer::Instance().GetTimeZone();
+    VerifyOrReturnValue(timeSynchronization != nullptr, false);
+
+    const auto & tzList = timeSynchronization->GetTimeZone();
     VerifyOrReturnValue(tzList.size() != 0, false);
     const auto & tz = tzList[0].timeZone;
+
     Events::TimeZoneStatus::Type event;
 
     event.offset = tz.offset;
     if (tz.name.HasValue())
     {
         event.name.SetValue(tz.name.Value());
     }
-    EventNumber eventNumber;
 
-    CHIP_ERROR error = LogEvent(event, ep, eventNumber);
-
-    if (CHIP_NO_ERROR != error)
+    std::optional<EventNumber> eventNumber;
+    if (eventsGenerator != nullptr)
+    {
+        eventNumber = eventsGenerator->GenerateEvent(event, ep);
+    }
+    if (eventNumber == std::nullopt)
```

Comment:
GenerateEvent already logs, so we do not need the extra log.
The only tricky part is if eventsGenerator is nullptr .... I wonder if we could centralize that part something like `bool VerifyAndLogIfNoEventGenerator()` and have code like `VerifyOrReturn(VerifyAndLogIfNoEventGenerator())` in our code or similar.